### DISCUSSION TITLE
serverless functions issue

### DISCUSSION THREAD

**Discussion by elvniv** [vercel/community#5011] [Help]
### Page to Investigate

api.klorah.app

### Steps to Reproduce

i am having a really frustrating time trying to understand the expected format that i need to host my server onto vercel.

i am working with and express server with multiple routes (one which is /hello for testing purposes) in my src/index i have an export 
module.exports = app;

 and in api/server i have this as my code : 
const express = require('express');
const app = require('../src/index');  // Adjust this path if your Express app's entry point is different

module.exports = app;
when i git push it runs build fine and reads "ready" on vercel but when i test in postman, i get 
The page could not be found

NOT_FOUND

i am not having fun

**Comment by amyegan** [comment]
Hey Elvin. Deploying Express apps to a serverless environment is a little more tricky that on a more traditional server setup. I recommend this guide as a good starting point: https://vercel.com/guides/using-express-with-vercel

**Comment by elvniv** [comment]
i followed the guide, restructured by repo, still having issues. could i please share my repo? its short

**Reply by elvniv** [reply]
please help me @amyegan  here is my repo https://github.com/Klorah/klorah-web-backend i did everything i could and this is holding us back

**Reply by amyegan** [reply]
It looks like your repo involves some additional resources, like Firebase, that prevent me from running it. 

But I know sometimes it helps to have an example for reference, so put together a small repo based on the guide I shared above: https://github.com/amyegan/express-vercel/tree/main

**Comment by elvniv** [comment]
is there any way you can tell my looking at the project structure? i feel
like making another mini repo is a waste of time respectfully.

**Reply by amyegan** [reply]
Your project has a lot going on, so I can’t pinpoint exactly what’s causing the problem just by quickly looking at the code. I’m not familiar with your project the way you are, and I don't know what error details you’re seeing. 

If you review my [example repo](https://github.com/amyegan/express-vercel/tree/main) and the [express.js guide](https://vercel.com/guides/using-express-with-vercel), particularly the [Drawbacks and Edge Cases](https://vercel.com/guides/using-express-with-vercel#drawbacks-and-edge-cases) section, that should help you identify how to get your project working.

**Comment by elvniv** [comment]
I thought there needed to be a public folder to host a server with vercel.
i notice that this repo doesnt have one. is that fine? could i reconfig to
make it not expect one


On Wed, Nov 29, 2023 at 6:11 PM Amy ***@***.***> wrote:

> Your project has a lot going on, so I can’t pinpoint exactly what’s
> causing the problem just by quickly looking at the code. I’m not familiar
> with your project the way you are, and I don't know what error details
> you’re seeing.
>
> If you review my example repo
> <https://github.com/amyegan/express-vercel/tree/main> and the express.js
> guide <https://vercel.com/guides/using-express-with-vercel>, particularly
> the Drawbacks and Edge Cases
> <https://vercel.com/guides/using-express-with-vercel#drawbacks-and-edge-cases>
> section, that should help you identify how to get your project working.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/vercel/discussions/5011#discussioncomment-7711311>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQUBODFMBPUFFOWKBMNCN53YG66ILAVCNFSM6AAAAAA763FUBSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TOMJRGMYTC>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

*- E*

**Comment by elvniv** [comment]
thsi is what happened when i cloned your example repo and replaced just the index code with mine while installing the packages i needed... i am a little lost at why i need a src directory and public if even your example doesnt have one.. do you have a different express github that actually hosted on vercel properly? this is not fun :)

Module not found: Error: Can't resolve './src' in '/vercel/path0'
Expand 19 Lines
12:40:20.643

<br class="Apple-interchange-newline">
